syllabus/curricula/plr/task_sampler.py

```diff
@@ -275,32 +297,66 @@ def _sample_unseen_level(self):
 
         return task
 
-    def compute_returns(self, gamma, gae_lambda, rewards, value_preds, masks):
+    def _evaluate_unseen_level(self):
+        sample_weights = self.unseen_task_weights / self.unseen_task_weights.sum()
+        task_idx = np.random.choice(range(self.num_tasks), 1, p=sample_weights)[0]
+        task = self.tasks[task_idx]
+
+        episode_data = self.evaluate_task(task, self.eval_envs, self.action_value_fn)
+        self.update_with_episode_data(episode_data)
+
+        self._update_staleness(task_idx)
+
+        return task
+
+    def compute_returns(self, rewards, values, masks, gamma, gae_lambda):
         assert self.requires_value_buffers, "Selected strategy does not use compute_rewards."
+
+        if isinstance(rewards, float):
+            rewards = [np.array(rewards)]
+
+        num_steps = len(rewards)
         gae = 0
-        returns = torch.zeros_like(rewards)
-        for step in reversed(range(rewards.size(0))):
-            delta = (
-                    rewards[step]
-                    + gamma * value_preds[step + 1] * masks[step + 1]
-                    - value_preds[step]
-            )
-            gae = delta + gamma * gae_lambda * masks[step + 1] * gae
-            returns[step] = gae + value_preds[step]
+        returns = np.zeros_like(rewards)
+        for step in reversed(range(num_steps)):
+            # Check if we are at the last step
+            if step == num_steps - 1:
+                delta = (rewards[step] - values[step])
+                gae = delta
+            else:
+                next_value = values[step + 1] if step + 1 < num_steps else 0
+                next_mask = masks[step + 1] if step + 1 < num_steps else 0
+                delta = (rewards[step]
+                         + gamma * next_value * next_mask
+                         - values[step]
+                         )
+                gae = delta + gamma * gae_lambda * next_mask * gae
+
+            gae_scal = gae[0] if isinstance(gae, np.ndarray) else gae
+            value_scal = values[step][0] if isinstance(values[step], np.ndarray) else values[step]
+
+            returns[step] = gae_scal + value_scal
```

Comment:
Why are you changing this function? You should not need to modify the GAE code, it's already correct.